gen.go

```diff
@@ -891,6 +1001,161 @@ func renameMutation(s string) string {
 	return s
 }
 
+func fieldCommentDescription(fieldType *types.InputValueDefinition) string {
+	oneLineDescription := strings.ReplaceAll(fieldType.Desc, "\n", " ")
+	if _, ok := fieldType.Type.(*types.NonNull); ok {
+		return fmt.Sprintf(" // %s (Required.)", oneLineDescription)
+	}
+	return fmt.Sprintf(" // %s (Optional.)", oneLineDescription)
+}
+
+func exampleStructTag(field *types.InputValueDefinition) string {
+	var exampleValue string
+	var unwrappedType types.Type
+
+	if nonNullType, ok := field.Type.(*types.NonNull); ok {
+		unwrappedType = nonNullType.OfType
+	} else {
+		unwrappedType = field.Type
+	}
+
+	switch fieldType := unwrappedType.(type) {
+	case *types.EnumTypeDefinition:
+		if enumValue, ok := enumExamples[strings.TrimSuffix(fieldType.String(), "!")]; ok {
+			exampleValue = enumValue
+		} else {
+			exampleValue = "ENUM_TODO"
+		}
+	case *types.InputObject:
+		return "" // omit 'example' struct tag, implicit nested tag is used
+	case *types.ScalarTypeDefinition:
+		if scalarValue, ok := scalarExamples[strings.TrimSuffix(fieldType.String(), "!")]; ok {
+			exampleValue = scalarValue
+		} else {
+			exampleValue = "SCALAR_TODO"
+		}
+	case *types.List:
+		if listValue, ok := listExamples[field.Name.Name]; ok {
+			exampleValue = listValue
+		} else {
+			exampleValue = "LIST_TODO"
+		}
+	case *types.TypeName:
+		exampleValue = "TYPENAME_TODO"
+	default:
+		exampleValue = "UNKNOWN_TODO"
+	}
+
+	return fmt.Sprintf(` example:"%s"`, exampleValue)
+}
+
+func yamlStructTag(fieldType *types.InputValueDefinition) string {
+	jsonStructTag := jsonStructTag(fieldType)
+	return strings.Replace(jsonStructTag, "json", "yaml", 1)
+}
+
+func jsonStructTag(fieldType *types.InputValueDefinition) string {
+	fieldName := fieldType.Name.Name
+	if isNullable(fieldType.Type) {
+		return fmt.Sprintf(`json:"%s,omitempty"`, fieldName)
+	}
+	return fmt.Sprintf(`json:"%s"`, fieldName)
+}
+
+func graphqlTypeToGolang(graphqlType string) string {
+	var convertedType string
+	nullable := "*Nullable["
+	if strings.HasSuffix(graphqlType, "!") {
+		graphqlType = strings.TrimSuffix(graphqlType, "!")
+	} else {
+		// GraphQL nullable --> Go pointer
+		convertedType += nullable
+	}
+
+	// GraphQL list --> Go slice
+	if strings.HasPrefix(graphqlType, "[") {
+		graphqlType = strings.TrimPrefix(graphqlType, "[")
+		graphqlType = strings.TrimSuffix(graphqlType, "]")
+		// NOTE: pretty sure we don't support slices containing pointers
+		graphqlType = strings.TrimSuffix(graphqlType, "!")
+		convertedType += "[]"
+	}
+
+	// GraphQL scalars/types --> Go type
+	switch graphqlType {
+	case "Boolean":
+		convertedType += "bool"
+	case "Float":
+		convertedType += "float64"
+	case "Int":
+		convertedType += "int"
+	case "ISO8601DateTime":
+		convertedType += "iso8601.Time"
+	case "String":
+		convertedType += "string"
+	default:
+		convertedType += graphqlType
```

Comment:
should we log a warning when this happens so we can be alerted to it?

Reply:
I don't think a warning is needed here. The `default` case catches non-scalar API Schema defined types like `IdentifierInput`, `UserIdentifierInput`, `ToolCategory`, `AlertSourceExternalIdentifier`, etc